synapse/storage/util/id_generators.py

```diff
@@ -548,7 +548,7 @@ def _add_persisted_position(self, new_id: int):
                 # do.
                 break
 
-    def _update_stream_positions_table_txn(self, txn):
+    def _update_stream_positions_table_txn(self, txn: LoggingTransaction):
```

Comment:
I'd recommend annotating this as `synapse.storage.types.Cursor` rather than `LoggingTransaction` unless you're actually relying on the extensions that are only in `LoggingTransaction`.